src/frontend/src/utils/countdown.ts

```diff
@@ -100,3 +101,87 @@ const calculateTimeRemaining = (
       .padStart(2, "0"),
   ];
 };
+
+// Prettify the number into a string: "mm:ss"
+function prettifySeconds(a: number): string {
+  const mins = Math.floor(a / 60)
+    .toString()
+    .padStart(2, "0");
+  const secs = Math.floor(a % 60)
+    .toString()
+    .padStart(2, "0");
+  return `${mins}:${secs}`;
+}
+
+/// A one shot countdown timer that emits the remaining time as a formatted string ("mm:ss")
+// through an AsyncIterable.
+//
+// The countdown cannot be restarted.
+export class AsyncCountdown {
```

Comment:
There seems to be an issue here somewhere, because I got this when I tried it on chrome (and no timer showing):
```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'expirationSeconds')
    at remainingSeconds (index.js:654:2560)
    at [Symbol.asyncIterator] (index.js:654:2720)
    at AsyncGenerator.next (<anonymous>)
    at index.js:654:966
    at Ci.update (index.js:654:1000)
    at Ci._$AS (index.js:610:2083)
    at k (index.js:2:385378)
    at S._$AI (index.js:2:386609)
    at G.p (index.js:2:386094)
    at S.$ (index.js:2:387284)
```

Reply:
Ah, eslint complained about me referencing `this` outside the object and I changed the code a bit just before pushing; I think that's the issue, thanks for catching it!

Reply:
Reverted the eslint-pleasing change and adapted the code to make it happy nonethless!